app/api-v1/services/orderService.js

```diff
@@ -0,0 +1,28 @@
+const { postOrderDb, getRecipeByIDs } = require('../../db')
+const { BadRequestError, IncorrectSupplierError } = require('../../utils/errors')
+
+async function postOrder(reqBody) {
+  // Will add a get function at a later date to check for duplication
+
+  // This section checks if the order manufacturer does not match the supplier
+  const uniqueRecipeIDs = [...new Set(reqBody.items)]
+  const recipes = await getRecipeByIDs(uniqueRecipeIDs)
+  if (recipes) {
+    recipes.forEach((recipeItem) => {
+      if (recipeItem.supplier != reqBody.supplier) {
+        throw new IncorrectSupplierError({ message: 'Order post error - Supplier does not match', service: 'order' })
+      }
+    })
+  }
```

Comment:
Note this won't catch recipes that don't exist. You may need to check that all `recipeIds` from the Set on ln8 are indeed present in this result. That will likely result in a new test case as well